### PR TITLE
Set `DistributedSampler` seed if `seed_everything` was called

### LINKED ISSUE #7016: seed for DistributedSampler
## 🐛 Bug
[torch.utils.data.distributed.DistributedSampler](https://pytorch.org/docs/stable/data.html#torch.utils.data.distributed.DistributedSampler) defaults to random seed `0` to shuffle the sampler if `shuffle=True`.
As lightning does not specify a seed, multi-gpu training with DistributedSampler in lightning will use a deterministic order independently from the user-specified seed (`seed_everything`).

https://github.com/PyTorchLightning/pytorch-lightning/blob/b85cfbe8f350a89c93ff967fa416859be7ebb4f3/pytorch_lightning/trainer/data_loading.py#L198-L202

https://github.com/PyTorchLightning/pytorch-lightning/blob/33cc9fe13838b3f97d0ee9a573900760aa6b63f4/pytorch_lightning/plugins/training_type/ddp.py#L91-L94

As a workaround, users can specify `replace_sampler_ddp` when creating the Trainer, but I think this is something that lightning should handle well out of the box.

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #7016 

By default our DistributedSampler is initialized with seed 0. This is fine, we need to set a seed there for distributed training because the samples need to be split between the GPUs/nodes. There is however currently no way to customize it unless the user supplies their own DistributedSampler. We can automate that by including the seed from `seed_everything`, if user calls it.

Note: For everyone that is using `seed_everything(x)` today, this feature will be affect reproducibility because the seed in the sampler will be set to x instead of 0. If this is undesired, we need to add a flag to `seed_everything` that can turn this off.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/data_loading.py [hunk 1]
```diff
@@ -197,7 +198,9 @@ def __init__(self, num_features, dataset, *args, **kwargs):
 
     def _get_distributed_sampler(self, dataloader, shuffle):
         kwargs = self.distributed_sampler_kwargs
-        kwargs['shuffle'] = shuffle and not self.overfit_batches
+        kwargs["shuffle"] = shuffle and not self.overfit_batches
+        if "seed" not in kwargs and "PL_GLOBAL_SEED" in os.environ:
+            kwargs["seed"] = int(os.getenv("PL_GLOBAL_SEED"))
```
Comment (carmocca) [inline pytorch_lightning/trainer/data_loading.py:203]:
```suggestion
        kwargs.setdefault("seed", int(os.getenv("PL_GLOBAL_SEED", 0)))
```

Comment (carmocca) [reply pytorch_lightning/trainer/data_loading.py:203]: Also, codecov is complaining about "lines #L201 - L203 were not covered by tests".

Can you double-check the tests are actually testing this?

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/trainer/data_loading.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/trainer/data_loading.py:203]: the test failed initially because I forgot the `int()` around the `os.getenv`. So, yes, the test touches these lines.

**Review by carmocca — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !
